flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/fakes/StorageClientFaker.java

```diff
@@ -120,18 +133,60 @@ public Job dryRunQuery(String projectId, String query) {
             };
         }
 
+        static class FaultyIterator<T> implements Iterator<T> {
+
+            private final Iterator<T> realIterator;
+            private final Double errorPercentage;
+            private final Random random = new Random();
```

Comment:
I wouldn't prefer using Random. If you absolutely have to, please use a fixed seed so that it's deterministic.

An alternative suggestion is to have it fail on every N-th invocation of `next()`.